src/connection.js

```diff
@@ -986,6 +1023,7 @@ class Connection extends EventEmitter {
       }
 
       this.request = request;
+      this.request.connection = this;
```

Comment:
The difference, at least as I see it, is the coupling.

In one case Request has knowledge of Connection class and internal method names inside the Connection class. For example if I rename the methods in the Connection class or move them to a sub-class I'd have to change the Request class.

In the other case, the Connection class is just passing the callbacks to Request and Request class honors the contract by invoking them, just as it invokes user callback on completion.